discord_thread_summarizer_using_openai/main.bal

```diff
@@ -0,0 +1,67 @@
+import ballerina/io;
+import ballerina/log;
+import ballerina/time;
+
+const TYPE_FORUM_CHANEL = 15;
+const time:Seconds ONE_DAY_IN_SECONDS = 86400;
+
+configurable int forumChannelId = ?;
+
+function readThreads(ActiveThreads activeThreads) returns error? {
+
+    foreach ChannelThread thread in activeThreads.threads {
+
+        if !check hasRecentMessages(thread) {
+            continue;
+        }
+
+        string prompt = check constructPrompt(thread);
+
+        string|error generateChatCompletionResult = generateChatCompletion(prompt);
+
+        if generateChatCompletionResult is string {
+            io:println(generateChatCompletionResult);
+        } else {
+            log:printError("Error occured when connecting to OpenAI: ",
+            generateChatCompletionResult,
+            stackTrace = generateChatCompletionResult.stackTrace());
+        }
+        io:println("--------------------------------------------------\n\n");
+    }
+}
+
+function constructPrompt(ChannelThread thread) returns string|error {
+    string prompt = string `${PROMPT}Thread URL: ${getThreadURL(thread.guild_id, thread.id)}${"\n"}Title: ${thread.name}${"\n"}Question: `;
+    Message[] allMessages = (check getMessages(thread.id)).reverse();
```

Comment:
I think the comment you had here previously was useful. We can keep it.

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/69/commits/99c8ac36bfdc145aa2fc590808d7e977da24f773